### PR TITLE
Refactor & several improvement.

### PR DESCRIPTION
**Pull request by ArthurSonzogni**

### CODE REVIEW

File: src/connman_handler/connman_handler.hpp [hunk 1]
```diff
@@ -133,7 +133,10 @@ class connman_h {
   void disconnect_wifi();
   void shell_helper(const char*);
 
-  /* Get a vector of names of wifi*/
+  // Get a vector of names of wifi
+  //
+  // TODO(satacker) Explain why calling this function is thread safe. Probably
+  // not thread safe. We probably should turn this into a pure function.
```
Comment (SAtacker) [inline src/connman_handler/connman_handler.hpp:139]: Turning it into a pure function would need a complete restructuring of the class . I think the `command_handler` should be merged with `connman_handler` and we can make the concerned impure functions into pure functions ?

**Comment by SAtacker** [comment]
What do you think of `beagle_config_utils` namespace instead of `connman_handler` , `command_handler` , which will help reduce redundant code? I would use the modularized UI implementations to implement basic stuff and the rest would go to `beagle_config_utils`

**Comment by ArthurSonzogni** [comment]
In my opinion, I wouldn't create the two directory: `connman_handler` and `command_handler`.

It seems to me you were about to separate your project by UI vs logic. I believe it is better to split by components and keep the UI and the logic of each component together. Those are usually tied, but components are independent. 

I would keep the logic of wireless and internet connexion sharing within their own component. Currently inside the directories:
- panel/wifi/
- panel/ics/

--

Adding many namespace isn't really useful. You just want to avoid your symbols not to clash with other libraries you depends on or are depending on you.
As long as your project isn't big enough, splitting your project with many different namespace doesn't look useful to me. This isn't a way to stay "organized".

---

Of course, those are just my opinions, feel free to proceed the way you like and iterate/change opinions over time your project evolve.

**Comment by SAtacker** [comment]
I totally agree with the recommendation you have. Yes, I was initially going with UI vs logic but it seems like I am building some redundant code like `shell_helpers`. I would implement the logic in the component itself as it's manageable in the same place as its UI component .

**Comment by SAtacker** [comment]
@ArthurSonzogni , I am stuck solving https://pastebin.com/6jSur9zE

**Pull request merged**